pkg/capabilities/utils.go

```diff
@@ -119,6 +119,7 @@ type TriggerAndId[T proto.Message] struct {
 // while adhering to the standard capability interface.
 func RegisterTrigger[I, O proto.Message](
 	ctx context.Context,
+	stop <-chan struct{},
```

Comment:
```suggestion
```
I think adding a return arg, like a `close func()`, would be a more idiomatic option. Makes things a little more intuitive for the callers. Although I can see that not every caller would be simplified so :shrug: